src/transform/src/join_implementation.rs

```diff
@@ -557,15 +569,46 @@ mod delta_queries {
             // A viable delta query requires that, for every order,
             // there is an arrangement for every input except for
             // the starting one.
-            if !orders
+            let arrangement_counts = orders
                 .iter()
-                .all(|o| o.iter().skip(1).all(|(c, _, _)| c.arranged))
-            {
-                return Err(TransformError::Internal(String::from(
-                    "delta plan not viable",
-                )));
+                .map(|o| o.iter().skip(1).filter(|(c, _, _)| c.arranged).count())
+                .collect::<Vec<_>>();
+            let expected_number_of_arrangements = inputs.len() - 1;
+            let missing_arrangements: std::collections::BTreeSet<usize> = orders
```

Comment:
Shouldn't the set key include the join key (in addition to the input number)? It's possible that different join paths want to arrange an input on different keys, and in that case a delta join will need more than 1 arrangement for that input. E.g., in ldbc q02 `message_hastag_tag`:
```
                    Join on=(#0{id} = #3{tagid} AND #2{messageid} = #5{messageid}) type=delta // { arity: 17 }
                      implementation
                        %0:l1 » %1:message_hastag_tag[#2]KA » %2:message[#1]KAiif
                        %1:message_hastag_tag » %2:message[#1]KAiif » %0:l1[#0]KA
                        %2:message » %1:message_hastag_tag[#1]KA » %0:l1[#0]KA
                      ArrangeBy keys=[[#0{id}]] // { arity: 1 }
                        Project (#0) // { arity: 1 }
                          Get l1 // { arity: 2 }
                      ArrangeBy keys=[[#1{messageid}], [#2{tagid}]] // { arity: 3 }
                        ReadIndex on=message_hastag_tag message_hastag_tag_messageid=[delta join lookup] message_hastag_tag_tagid=[delta join lookup] // { arity: 3 }
                      ArrangeBy keys=[[#1{messageid}]] // { arity: 13 }
                        ReadIndex on=message message_messageid=[delta join lookup] // { arity: 13 }
```

Reply:
Oh, good catch... thank you!